### PR TITLE
Optical Distortion main changes and augmentation parameter reflection

### PR DESCRIPTION
**Pull request by erezposner**
I've made some modification to optical distort.
your current implementation doesn't use real distortion in the sense of true polynomial with 5 different values. I've modified it.

in  transforms_interface.py - added ability to reflect parameters out. I think it is needed if people would want to get the parameter used in the augmentation.

also, I've added to opticalDistortion keypoints distortion that wasn't implemented.

### CODE REVIEW

**Comment by MichaelMonashev** [comment]
@erezposner , you need split your transform to 2 parts. Move this code https://github.com/albumentations-team/albumentations/pull/571/files#diff-a80a9cab03e62b28e741d05ef8182abcR1170 to functional.py . After moving you can use it without `fix_params`.

File: albumentations/augmentations/functional.py [hunk 1]
```diff
@@ -1084,14 +1084,15 @@ def optical_distortion(
     height, width = img.shape[:2]
 
     fx = width
-    fy = height
+    fy = width
```
Comment (Dipet) [inline albumentations/augmentations/functional.py:1087]: For what reason you changed height to width?

File: albumentations/augmentations/functional.py [hunk 2]
```diff
@@ -1084,14 +1084,15 @@ def optical_distortion(
...
 
     cx = width * 0.5 + dx
     cy = height * 0.5 + dy
 
     camera_matrix = np.array([[fx, 0, cx], [0, fy, cy], [0, 0, 1]], dtype=np.float32)
 
-    distortion = np.array([k, k, 0, 0, 0], dtype=np.float32)
+    # distortion = np.array([k, k, 0, 0, 0], dtype=np.float32)
+    distortion = np.array(k, dtype=np.float32)
```
Comment (Dipet) [inline albumentations/augmentations/functional.py:1095]: What the reason?

File: albumentations/augmentations/functional.py [hunk 3]
```diff
@@ -1101,6 +1102,12 @@ def optical_distortion(
         borderMode=border_mode,
         borderValue=value,
     )
+    # map1, map2 = cv2.initUndistortRectifyMap(camera_matrix, distortion, None, None, (width, height), cv2.CV_32FC1)
+    # for i in range(img.shape[0]):
+    #     img[i,...] = cv2.remap(
+    #         img[i,...], map1, map2, interpolation=interpolation, borderMode=border_mode, borderValue=value
+    #     )
+    # return img
```
Comment (Dipet) [inline albumentations/augmentations/functional.py:1110]: Need to remove

File: albumentations/augmentations/transforms.py [hunk 4]
```diff
@@ -432,10 +433,37 @@ def __init__(
         self.border_mode = border_mode
         self.value = value
         self.mask_value = mask_value
+        self.fix_params = fix_params
+        if self.fix_params:
+            self.k = [random.uniform(self.distort_limit[0], self.distort_limit[1]) for i in range(5)]
+            self.dx = round(random.uniform(self.shift_limit[0], self.shift_limit[1]))
+            self.dy = round(random.uniform(self.shift_limit[0], self.shift_limit[1]))
```
Comment (Dipet) [inline albumentations/augmentations/transforms.py:440]: Why we need this?

File: albumentations/augmentations/transforms.py [hunk 5]
```diff
@@ -432,10 +433,37 @@ def __init__(
...
+
+        # distortion = np.array([ params['params']['k'],  params['params']['k'], 0, 0, 0], dtype=np.float32)
+        k = params["params"]["k"]
+        distortion = np.array(k, dtype=np.float32)
+
+        new_points = cv2.undistortPoints(keypoints, camera_matrix, distortion, P=camera_matrix)
+
+        return tuple(new_points)
+        # keypoints = cv2.remap(
+        #   keypoints, map1, map2, interpolation=interpolation, borderMode=self.border_mode, borderValue=self.value
```
Comment (Dipet) [inline albumentations/augmentations/transforms.py:464]: Need to remove comments

File: albumentations/core/transforms_interface.py [hunk 6]
```diff
@@ -111,7 +111,7 @@ def apply_with_params(self, params: Dict[str, Any], **kwargs) -> Dict[str, Any]:
                 res[key] = target_function(arg, **dict(params, **target_dependencies))
             else:
                 res[key] = None
-        return res
+        return res,params
```
Comment (Dipet) [inline albumentations/core/transforms_interface.py:114]: We have `ReplayCompose` for this logic.

**Review by Dipet — CHANGES_REQUESTED** [review]

**Pull request closed without merging**